assets/src/stories-editor/components/resizable-box/index.js

```diff
@@ -190,14 +192,26 @@ class EnhancedResizableBox extends Component {
 						}
 
 						// If the applied measures get too small for text, use the previous measures instead.
-						const scrollWidth = textElement.scrollWidth;
-						const scrollHeight = textElement.scrollHeight;
-						// If the text goes over either of the edges, stop resizing from both sides
-						// since the text is filling in the room from both sides at the same time.
+						let scrollWidth = textElement.scrollWidth;
+						let scrollHeight = textElement.scrollHeight;
+
 						if ( appliedWidth < scrollWidth || appliedHeight < scrollHeight ) {
-							reachedMinLimit = true;
-							appliedWidth = lastWidth;
-							appliedHeight = lastHeight;
+							// If we have amp-fit-text on, then reduce the font size to prevent the block
+							// shifting due to font size not fitting.
+							if ( ampFitTextElement ) {
+								while ( appliedWidth < scrollWidth || appliedHeight < scrollHeight ) {
```

Comment:
Looks like the sliding issue came from the initial pre-resize amp-fit-text font size being too large for the container and this causing the container to "slide". After resizing stopping and the font calculation happened, the block "jumped" to its correct position. Note that this did not influence the top/left positioning in any way.

WDYT of this approach? Reducing the font size already while resizing.

![resizing](https://user-images.githubusercontent.com/3294597/66335315-1b304000-e93b-11e9-8b14-4cecb095df87.gif)

Reply:
If it's performant, it's a great idea and better UX than the original. Does it work well for even long texts and weird fonts (with many ligatures)?

Reply:
It only works when making the block smaller, not when making it larger, which is confusing. Also doesn't seem to respect `MAX_FONT_SIZE` and `MIN_FONT_SIZE`. I just made a block with lots of text very small and the text basically disappeared because the font size was set to `0px`.

Reply:
It definitely should have font limits. Also, if we go that way we should also implement the same logic for the post meta blocks as well to make it consistent.

On performance: seems to work well with weird fonts as well, at least as far as I tested.

On working when making the block smaller only: yep, I was doubting that it might be confusing. When turning off amp-fit-text and the previously saved font size is too large then we automatically increase the size of the block so that the text would fit inside. However, we do not reduce the block if there's a lot of room. Meaning that elsewhere we do care about the font not going over the lines, not so much about there being extra room left. Because of that, I was 50-50 that perhaps it's OK just to make sure that the text fits the block but ignore if there's more room. It's a bit different case for amp-fit-text though.

Anyway -- I'll add the size increasing as well now that you brought it out as well.